discord_connector.py

```diff
@@ -0,0 +1,566 @@
+        summary['num_channels'] = num_channels
+
+        return status
+
+
+    def _handle_send_message(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        destination = param['destination']
+        message = param['message']
+
+        status, channel = self.run_in_loop(self._guild.fetch_channel(destination), action_result, message = "Cannot fetch channel from Discord.")
+        status, message = self.run_in_loop(channel.send(message), action_result,
+                                               message="Cannot send message to Discord.")
+
+        action_result.add_data({
+            "message_id": message.id
+        })
+
+        return status
+
+
+    def _handle_kick_user(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        user_id = param['user_id']
+        reason = param['reason']
+
+        status, user = self.run_in_loop(self._guild.fetch_member(user_id), action_result,
+                                        message="Cannot fetch member from Discord.")
+        status, result = self.run_in_loop(self._guild.kick(user, reason=reason), action_result,
+                                          message="Cannot kick the user from Discord.")
+
+        return status
+
+
+    def _handle_ban_user(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        user_id = param['user_id']
+        reason = param['reason']
+        delete_message_seconds = param['delete_message_seconds']
+
+        status, user = self.run_in_loop(self._guild.fetch_member(user_id), action_result, message = "Cannot fetch member from Discord.")
+        status, result = self.run_in_loop(self._guild.ban(user, reason=reason, delete_message_seconds=delete_message_seconds), action_result, message = "Cannot ban the user from Discord.")
+
+        return status
+
+    def _handle_get_user(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        user_id = param['user_id']
+
+        status, user = self.run_in_loop(self._guild.fetch_member(user_id), action_result, message = "Cannot fetch member from Discord.")
+
+        action_result.add_data({
+            "display_name": user.display_name,
+            "name": user.name,
+            "created_at": str(user.created_at),
+            "system": user.system,
+            "public_flags": user.public_flags.all()
+        })
+
+
+        return status
+
+    def run_in_loop(self, coroutine, action_result, message = ""):
```

Comment:
Maybe we could rename `message` to `error_message` to make it  more inline with variable purpose